_drafts/reproducible-firmware-builds.md

```diff
@@ -0,0 +1,611 @@
+---
+title: "Reproducible Firmware Builds"
+description: "A discussion about reproducible builds, why they matter and steps you can take to make your firmware build reproducible"
+author: chris
+---
+
+If you have ever worked on a large scale embedded project before, you've probably run into situations where the binary behaves differently depending on where it was compiled. For example, maybe the binary completely fails to compile on one computer and on another it compiles but crashes on boot!
+
+<!-- excerpt start -->
+
+In this article we will discuss what a **Reproducible Build** is, walk through the process of updating a firmware project so the build _is_ reproducible and explore how we can leverage what we have learned for other aspects of development.
+
+<!-- excerpt end -->
+
+{:.no_toc}
+
+## Table of Contents
+
+<!-- prettier-ignore -->
+* auto-gen TOC:
+{:toc}
+
+## Reproducible Builds
+
+A build is said to be "Reproducible" when given a set of source code, the exact same binary can be generated by different people on different computers. Many open source projects (such as Debian GNU/Linux) are working towards making the entire build reproducible[^8].
+
+### Why does it matter if the build is Reproducible?
+
+Common problems that arise when a build is not reproducible is the binary behaving differently for developers on the team. Debugging and supporting these kinds of issues can be a significant time sink. It's hard to debug a problem when you can't reproduce it yourself! Making builds reproducible can eliminate this class of problem and is the primary reason I reason I think reproducible builds are useful for embedded projects. We'll explore an example of this type of problem in the next section.
+
+Having reproducible builds can also be helpful when trying to recreate the debug symbols for an image where the symbols were not preserved or to be able to re-generate a build that hasn't been updated in a while (i.e an image used in the factory).
```

Comment:
latin nit: "i.e." stands for "id est" which means "in other words". The latin abbreviation you're looking for here is "e.g." which stands for "exempli gratia" i.e. "for example" (see what I did there?)